packages/nouns-contracts/contracts/Rewards.sol

```diff
@@ -0,0 +1,280 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.19;
+
+import { INounsDAOLogicV3 } from './interfaces/INounsDAOLogicV3.sol';
+import { INounsAuctionHouseV2 } from './interfaces/INounsAuctionHouseV2.sol';
+import { NounsDAOV3Types } from './governance/NounsDAOInterfaces.sol';
+import { ERC721 } from '@openzeppelin/contracts-v5/token/ERC721/ERC721.sol';
+import { IERC20 } from '@openzeppelin/contracts-v5/token/ERC20/IERC20.sol';
+import { Ownable } from '@openzeppelin/contracts-v5/access/Ownable.sol';
+
+contract Rewards is ERC721('NounsClientIncentives', 'NounsClientIncentives'), Ownable {
+    INounsDAOLogicV3 public immutable nounsDAO;
+    INounsAuctionHouseV2 public immutable auctionHouse;
+
+    uint32 public nextTokenId = 1;
+
+    uint256 public lastProcessedAuctionId;
+    uint32 public nextProposalIdToReward;
+    uint256 public nextProposalRewardFirstAuctionId;
+    uint256 lastProposalRewardsUpdate = block.timestamp;
+
+    RewardParams public params;
+
+    IERC20 public ethToken; // TODO make configurable
+
+    mapping(uint32 clientId => uint256 balance) public clientBalances;
+
+    struct ClientData {
+        address payoutWallet;
+    }
+
+    struct RewardParams {
+        uint32 minimumRewardPeriod;
+        uint8 numProposalsEnoughForReward;
+        uint16 proposalRewardBps;
+        uint16 votingRewardBps;
+        uint16 auctionRewardBps;
+        uint16 proposalEligibilityQuorumBps;
+    }
+
+    constructor(
+        address owner,
+        address nounsDAO_,
+        address auctionHouse_,
+        address ethToken_,
+        uint32 nextProposalIdToReward_,
+        uint256 lastProcessedAuctionId_,
+        uint256 nextProposalRewardFirstAuctionId_,
+        RewardParams memory rewardParams
+    ) Ownable(owner) {
+        nounsDAO = INounsDAOLogicV3(nounsDAO_);
+        auctionHouse = INounsAuctionHouseV2(auctionHouse_);
+        ethToken = IERC20(ethToken_);
+        nextProposalIdToReward = nextProposalIdToReward_;
```

Comment:
can we deduce this by reading the current prop ID from the DAO and maybe incrementing by one?

Reply:
yeah we can do that.
what's the benefit? less parameters?

Reply:
yes, less room for human error :)